app/views/admin/attachments/_form.html.erb

```diff
@@ -23,23 +23,39 @@
 
   <% if attachment.is_a?(HtmlAttachment) %>
     <%= form.fields_for :govspeak_content do |govspeak_fields| %>
+      <div class="govuk-!-margin-bottom-8">
+        <%= render "govuk_publishing_components/components/checkboxes", {
+          name: "attachment[govspeak_content_attributes][manually_numbered_headings]",
+          items: [
+            {
+              label: "Manually numbered headings",
+              value: "1",
+              checked: form.object.content_id != nil
```

Comment:
Is this the right condition for the checkbox to be checked? It looks like this will mean the box is always checked, so long as there's a content ID.

But I was expecting to see something more like this:

```
checked: form.object.govspeak_content.manually_numbered_headings
```

i.e. reading the value of the `manually_numbered_headings` field, not `content_id`.

Reply:
Yeah, this makes very little sense to me but I believe it's correct. The way Nikin described it to me (and the way it works currently) is that the box is checked only once the document has been saved and you return to edit it. No-one seems to know how this is useful.

Reply:
@davidtrussler I don't think it works as you've described – I have a feeling some wires have become crossed somewhere. It might be easier to talk this through together.

Looking through the code for the Bootstrap version of this page:
https://github.com/alphagov/whitehall/blob/049dac792091d3de01b479805ecf269cd50a4dd5/app/views/admin/attachments/_form_legacy.html.erb#L17-L26

It looks like there are a couple of things going on here:

1. The checkbox behaves like a 'normal' checkbox – i.e. it's 'checked' state is controlled by the user
    - if the user checks it & hits save, it'll stay checked when the page is reloaded
    - if they leave it unchecked & hit save, it'll remain unchecked
    - it defaults to being unchecked, and it'll never automatically check itself
2. There's some inline help which only shows when the checkbox is checked – i.e. the `<div class="js-manual-numbering-help...` is hidden if the user hasn't checked the checkbox.
    - this behaviour is implemented server-side with a conditional class:
      ```ruby
      <%= ' js-hidden' unless attachment.manually_numbered_headings? %>
      ```
    - JavaScript exists to implement this behaviour client-side, **but it's broken**.
      It targets `#attachment_manually_numbered_headings` but the correct ID to target is `#attachment_govspeak_content_attributes_manually_numbered_headings`
      https://github.com/alphagov/whitehall/blob/049dac792091d3de01b479805ecf269cd50a4dd5/app/assets/javascripts/admin/views/attachments/_form.js#L25-L29

You said:
> The way Nikin described it [...] is that the box is checked only once the document has been saved and you return to edit it.

I suspect the broken show/hide JavaScript is what led to this description – i.e. the inline help only appears after you've saved and returned to edit the HTML attachment. (Because that's when the Ruby conditional would kick in.)

But I think that's a separate issue to the first piece of behaviour – i.e. the checkbox being manually controlled by a user, rather than being automatically checked after the first save.

**TL;DR:** I still think this is the desired behaviour:

```suggestion
              checked: form.object.govspeak_content.manually_numbered_headings
```

Reply:
@ollietreend I've updated this section in [this commit](https://github.com/alphagov/whitehall/pull/6856/commits/625dd90fb39debef67aa7cda18c4b66fc9bbc1ac). There is though some weirdness going on with the checkbox component that won't show conditional items if there is no header or hint text. Whether this is a problem with the component or how it is supposed to work by design I'm not sure. 

Consequently I've implemented it as a kind of mash-up of the design, which I will also run by JP for comment as it may or may not be acceptable: for instance I think we might want different wording for "Guidance on manually numbered headings" if we go with this. 

Shown in the screenshots below compared to the supplied design. 

| Design | Unchecked | Checked |
| --- | --- | --- |
| ![Screenshot 2022-10-10 at 13 39 33](https://user-images.githubusercontent.com/6080548/194880950-b1011e95-28ea-4077-9eac-076d7f8af384.png) | ![Screenshot 2022-10-10 at 14 51 38](https://user-images.githubusercontent.com/6080548/194882064-c070953e-5da6-4506-a291-7e76b1043562.png) | ![Screenshot 2022-10-10 at 14 49 09](https://user-images.githubusercontent.com/6080548/194881655-f3b613dc-f20b-44fb-adff-217b942d9d0a.png) |

Reply:
@davidtrussler Nice use of the conditional reveal functionality to avoid us having to write our own show/hide JS.

The need for a heading/hint is annoying – I guess it's because we're sort-of hacking the checkboxes component into serving our needs. Usually a heading would be required because the user would be choosing from a list of multiple items.

Just wondering: have you already tried setting the options `visually_hide_heading: true, no_hint_text: true`? These might do what we need [according to the docs](https://components.publishing.service.gov.uk/component-guide/checkboxes). But I'm not sure if they can both be used together?

Failing that, I wonder if we can get the best of both worlds by simply extracting the HTML we need from that component. Then we could hardcode the HTML into the view without using the partial – but still benefit from the conditional reveal functionality provided by JS.

What do you think?

Reply:
Hello @ollietreend, as you are aware, the outcome of discussions with Carlos & JP (& yourself) is that we should go with this as above. 

Also, I'm going to move this ticket back to "In development" for the minute (and make this PR a draft) in order to deal with the right hand sidebar after the discussion earlier about using a different component in that.